include/deal.II/base/symmetric_tensor.templates.h

```diff
@@ -42,7 +42,9 @@ std::array<Number,2>
 eigenvalues (const SymmetricTensor<2,2,Number> &T)
 {
   const Number upp_tri_sq = T[0][1]*T[0][1];
-  if (upp_tri_sq == Number(0.0))
+  // TODO [JPP]: This is not the correct way to do branching with Adol-C
+  // taped doubles with advanced branching is enabled.
+  if (numbers::is_zero_valued(upp_tri_sq))
```

Comment:
Is this (and the places below) the right place to leave this TODO? It seems pretty unrelated to the purpose of the function. If you think that there is something wrong with this sequence here, maybe the right place for the TODO is inside `is_zero_valued()`...